File: c/LeetCode_169_MajorityElement.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int majorityElement(int* nums, int numsSize) {
    int ind = 0;
    int counter = 0;
    
    for (int i = 0; i < numsSize; i++) {
        if (nums[ind] == nums[i]) {
            ++counter;
        } else {
            --counter;
        }

        if (counter == 0) {
            ind = i;
            counter = 1;
        }
    }

    return nums[ind];
}


int main() {
    int *nums = malloc(3 * sizeof(int));
    nums[0] = 3;
    nums[1] = 3;
    nums[2] = 4;

    majorityElement(nums, 3);
    return 0;
}

File: python/LeetCode_169_MajorityElement.py
def majorityElement(nums) -> int:
    inx = 0
    counter = 0    
    for i in range(len(nums)):
        if nums[i] == nums[inx]:
            counter += 1
        else:
            counter -= 1

        if counter == 0:
            inx = i
            counter = 1

    return nums[inx]


print(majorityElement([2,1,2]))



"""

from collections import Counter

def majorityElement(nums) -> int:
    c = Counter(nums)
    mmax = 0
    res = 0 
    for key, val in c.items():
        if val > mmax:
            mmax = val 
            res = key
    
    return res


print(majorityElement([3,2,3]))

"""